src/utils/withWidth.js

```diff
@@ -0,0 +1,120 @@
+// @flow weak
+
+import React, { Component, PropTypes } from 'react';
+import EventListener from 'react-event-listener';
+import createHelper from 'recompose/createHelper';
+import createEagerFactory from 'recompose/createEagerFactory';
+
+const BREAKPOINTS = [
+  'xs',
+  'sm',
+  'md',
+  'lg',
+  'xl',
+];
+
+export const isWidthUp = (baseWidth, width) => (
+  BREAKPOINTS.indexOf(baseWidth) <= BREAKPOINTS.indexOf(width)
+);
+
+export const isWidthDown = (baseWidth, width) => (
+  BREAKPOINTS.indexOf(baseWidth) > BREAKPOINTS.indexOf(width)
+);
+
+function withWidth(options = {}) {
+  const {
+    resizeInterval = 166, // Corresponds to 10 frames at 60 Hz.
+  } = options;
+
+  return (BaseComponent) => {
+    const factory = createEagerFactory(BaseComponent);
+
+    return class WithWidth extends Component {
+      static contextTypes = {
+        theme: PropTypes.object.isRequired,
+      };
+
+      state = {
+        width: null,
+      };
+
+      componentDidMount() {
+        this.updateWidth(window.innerWidth);
+      }
+
+      componentWillUnmount() {
+        clearTimeout(this.deferTimer);
+      }
+
+      deferTimer = null;
+
+      handleResize = () => {
+        clearTimeout(this.deferTimer);
+        this.deferTimer = setTimeout(() => {
+          this.updateWidth(window.innerWidth);
+        }, resizeInterval);
+      };
+
+      updateWidth(innerWidth) {
+        const breakpoints = this.context.theme.breakpoints;
+        let width = null;
+
+        /**
+         * Start with the slowest value as low end devices often have a small screen.
+         *
+         * innerWidth |0      xs      sm      md      lg      xl
+         *            |-------|-------|-------|-------|-------|------>
+         * width      |  xs   |  xs   |  sm   |  md   |  lg   |  xl
+         */
+        let index = 1;
+        while (width === null && index < breakpoints.keys.length) {
+          const currentWidth = breakpoints.keys[index];
```

Comment:
Oups, the spec says that `Object.keys` as the same order than `for...in`. That order is arbitrary 🙈.